controllers/stargate/stargate_controller.go

```diff
@@ -195,7 +195,14 @@ func (r *StargateReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	}
 
 	// Compute the desired deployments
-	desiredDeployments := stargateutil.NewDeployments(stargate, actualDc)
+	desiredDeployments := stargateutil.NewDeployments(stargate, actualDc, logger)
```

Comment:
Not a big deal, but maybe you could reconcile the Vector config map first, then call this function. This way if there is an error with the former, we don't lose time calling the latter.